libfwupdplugin/fu-pefile-firmware.c

```diff
@@ -118,28 +139,82 @@ fu_pefile_firmware_parse(FuFirmware *firmware,
 	for (guint idx = 0; idx < nr_sections; idx++) {
 		guint32 sect_length = 0x0;
 		guint32 sect_offset = 0x0;
-		guint8 name[8] = {0x0};
+		guint32 zeroes = 0x0;
 		g_autofree gchar *sect_id = NULL;
 		g_autoptr(FuFirmware) fw_sect = NULL;
 		g_autoptr(GBytes) sect_blob = NULL;
 
-		/* read the section name */
-		if (!fu_memcpy_safe(name,
-				    sizeof(name),
-				    0x0,
-				    buf,
-				    bufsz,
-				    hdr_offset,
-				    sizeof(name),
-				    error))
-			return FALSE;
-		sect_id = fu_strsafe((const gchar *)name, sizeof(name));
-		if (sect_id == NULL) {
-			g_set_error_literal(error,
-					    FWUPD_ERROR,
-					    FWUPD_ERROR_INVALID_FILE,
-					    "no section name");
+		if (!fu_memread_uint32_safe(buf,
+					    bufsz,
+					    hdr_offset,
+					    &zeroes,
+					    G_LITTLE_ENDIAN,
+					    error))
 			return FALSE;
+
+		if (zeroes == 0) {
+			guint32 str_idx = 0x0;
+
+			if (!fu_memread_uint32_safe(buf,
+						    bufsz,
+						    hdr_offset + 4,
+						    &str_idx,
+						    G_LITTLE_ENDIAN,
+						    error))
+				return FALSE;
+
+			sect_id = fu_strsafe((const gchar *)buf + string_table + str_idx,
+					     bufsz - string_table);
+			if (sect_id == NULL) {
+				g_set_error_literal(error,
+						    FWUPD_ERROR,
+						    FWUPD_ERROR_INVALID_FILE,
+						    "no section name");
+				return FALSE;
+			}
```

Comment:
Is this COFF? I didn't see it https://learn.microsoft.com/en-us/windows/win32/debug/pe-format here

Reply:
Ohh it's PE COFF -- i just wondered where you got that spec. It might be correct!

Reply:
Ahh, so this alternate zeros encoding is from https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#coff-symbol-table -- but that's only for the symbol table, not the section table. I'll fix things up.